### ISSUE TITLE
External Markdown with math equations broken in 3.8.0

### ISSUE THREAD

**Issue by mthmulders** [hakimel/reveal.js#2507]
In my Reveal.js presentations I have some math equations. I prefer to have my slides in external Markdown files.

Now, in Reveal.js 3.6.0 this worked great, but in 3.8.0 it doesn't work anymore. The equations have a red backslash, some markup is not processed (showing \_note\_ instead of _note_) and the animations are out of order.

Initially, 3.6.0 renders as
<img width="773" alt="reveal360-initial" src="https://user-images.githubusercontent.com/430114/66702023-3a481c80-ed03-11e9-987e-8fe2fd52983c.png">

But in 3.8.0, this becomes
<img width="773" alt="reveal380-initial" src="https://user-images.githubusercontent.com/430114/66702034-48963880-ed03-11e9-9159-4f88fa802c9c.png">

In the end state, 3.6.0 renders as
<img width="773" alt="reveal360-final" src="https://user-images.githubusercontent.com/430114/66702037-521fa080-ed03-11e9-8db5-92b4f7af02f2.png">

Whereas 3.8.0 renders as 
<img width="772" alt="reveal380-final" src="https://user-images.githubusercontent.com/430114/66702039-58ae1800-ed03-11e9-9756-23d95849d36a.png">

I've created a [repo](https://github.com/mthmulders/reveal.js-markdown-mathjax) with both versions in it.

**Comment by mthmulders** [comment]
Seems to have been introduced with 3.8.0, as 3.7.0 works just like 3.6.0.

**Comment by mthmulders** [comment]
Adding the slidechanged or fragmentshown event handlers (as described in #226 resp #505) doesn't help.

**Comment by burgerga** [comment]
I don't know why it changed, but it is preprocessed by markdown. You might need an extra escape so `\\*` instead of `\*`. The other suggestions I came across is placing the equation in backticks but that might not work.

**Comment by burgerga** [comment]
> I don't know why it changed, but it is preprocessed by markdown. You might need an extra escape so `\\*` instead of `\*`. The other suggestions I came across is placing the equation in backticks but that might not work.

Never mind, it's actually the other way around, you don't need the backslashes anymore. Just a `*` should work, so



(Don't know why it changed though, it's not something in the math plugin, I guess something in markdown.js)

Very strange how in 

```
1. $(d \* 3) − 1  \mod (10 \* 16) = 0$
1. <!-- .element: class="fragment" --> $(d \* 3) − 1  \mod (10 \* 16) = 0$
```

the top line works, but not the bottom.

**Comment by mthmulders** [comment]
Thanks for your reply, @burgerga. Removing the backslash indeed fixes rendering of the formula.

The other issue still remains: markup that is part of a fragment is not applied. So where you would have

```md
<!-- .element: class="fragment" -->This is an _example_ fragment.
```

You would expect the word "example" to emphasised, e.g.`This is an <em>example</em> fragment.` Instead, it becomes `This is an _example_ fragment.`

**Comment by mthmulders** [comment]
I've added Reveal 3.9.2 to the [repo](https://github.com/mthmulders/reveal.js-markdown-mathjax).

**Comment by Brickaction** [comment]
7.3.11

**Comment by mthmulders** [comment]
I'm not sure what that means, @Brickaction....

**Comment by burgerga** [comment]
> Thanks for your reply, @burgerga. Removing the backslash indeed fixes rendering of the formula.
> 
> The other issue still remains: markup that is part of a fragment is not applied. So where you would have
> 
> ```lisp
> This is an _example_ fragment.
> ```
> 
> You would expect the word "example" to emphasised, e.g.`This is an <em>example</em> fragment.` Instead, it becomes `This is an _example_ fragment.`

Yes, I guess this is the real issue, so it's not MathJax related at all actually. The markdown is simply not processed anymore (if markdown would be processed, the backslash in `\*` would have been removed, which is why `\*` works outside of fragments). 
Maybe you can change the title of your bug. 

@hakimel do you have any idea what changed? (also happens in v4)

**Comment by mthmulders** [comment]
I'm not sure I correctly understand... Something like "Markdown is not processed inside Reveal fragments" ?

**Comment by burgerga** [comment]
Yes almost, to be more specific "Markdown processed incorrectly inside Markdown fragments added with element attributes", because if you modify https://revealjs.com/markdown/#element-attributes to
```
- Item 1 <!-- .element: class="fragment"  -->
- Item 2 test <!-- .element: class="fragment"  -->
- Item 3 _test_ <!-- .element: class="fragment"  -->
- <!-- .element: class="fragment"  --> Item 4 _test_ 
```
this also gives very unexpected results.

**Comment by gitpitch** [comment]
I wonder if the issue was introduced when the revealjs dependency on _marked_ was updated to v0.6.0. That update happened in reveal 3.8.0. Possibly as a result of [this change](https://github.com/markedjs/marked/pull/1383) to marked. It might be worth dropping in the older [v0.5.2](https://github.com/markedjs/marked/releases/tag/v0.5.2) library to see if the upgrade to v0.6.0 is indeed the culprit.

**Comment by mthmulders** [comment]
Good catch, @gitpitch. I've changed my repo to use Reveal 3.8.0 with the Marked version that shipped with 3.7.0. Reverting the backslash-change makes the slide behave as before.

**Comment by burgerga** [comment]
Just the `marked.js` or also the `markdown.js`? Because when I try to use marked 0.5.2 in reveal 4.0.0 it still doesn't work for me 😕

**Comment by mthmulders** [comment]
> Just the `marked.js` or also the `markdown.js`? Because when I try to use marked 0.5.2 in reveal 4.0.0 it still doesn't work for me 😕

That was with marked.js and markdown.js from Reveal 3.7.0.

> 
> EDIT:
> does this also work correctly for you with marked 0.5.2?

With Reveal 3.8.0 and both marked.js and markdown.js from Reveal 3.7.0, I see appearing (in order):

* Item 3
* Item 1
* Item 2 test
* _test_ behind Item 3
* Item 4 _test_

**Comment by Brickaction** [comment]
> I'm not sure what that means, @Brickaction....

Neither do I but I feel like those numbers are important

**Comment by burgerga** [comment]
Reported as troll.

**Comment by burgerga** [comment]
> That was with marked.js and markdown.js from Reveal 3.7.0.

Ah, so the issue is not `marked.js` but rather the `markdown.js` plugin which uses `marked.js`. I quickly had a look at it, but between 3.7.0 and 3.8.0 there was a substantial rewrite of the plugin (https://github.com/hakimel/reveal.js/compare/3.7.0...3.8.0#diff-7bb64d29f19e738b6dfb4b72976507c4), and I'm completely unfamiliar with that code, so I hope @hakimel will step in and have a look at it :wink:

> With Reveal 3.8.0 and both marked.js and markdown.js from Reveal 3.7.0, I see appearing (in order):
> 
> * Item 3
> * Item 1
> * Item 2 test
> * _test_ behind Item 3
> * Item 4 _test_

This is then the second issue :disappointed:, which has probably never worked as intended
So the fragment element attribute should be added right after the `*`/`-` to make it work properly with markdown in the items, so instead of 
```
<section data-markdown>
  <script type="text/template">
    - Item 1 <!-- .element: class="fragment" data-fragment-index="2" -->
    - Item 2 <!-- .element: class="fragment" data-fragment-index="1" -->
  </script>
</section>
```

[the documentation](https://revealjs.com/markdown/#element-attributes) should probably read

```
<section data-markdown>
  <script type="text/template">
    - <!-- .element: class="fragment" data-fragment-index="2" --> Item 1
    - <!-- .element: class="fragment" data-fragment-index="1" --> Item 2
  </script>
</section>
```
to circumvent this.

**Comment by nkutsche** [comment]
I'm not sure if it is clear, but a nasty workaround is to use HTML markup in Markdown:

```md
- <!-- .element: class="fragment" data-fragment-index="1" --> <em>Item 1</em>
- <!-- .element: class="fragment" data-fragment-index="1" --> _Item 2_
```

The first item is rendered correctly, the second is not.
